### PR TITLE
elvish: Update to 0.8 and change build procedure.

### PR DESCRIPTION
**Pull request by xiaq**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR changes the building procedure, so I assume that the "fix" workflow applies (as opposed to the "upgrade" workflow).

### CODE REVIEW

File: Formula/elvish.rb [hunk 1]
```diff
@@ -20,8 +20,6 @@ def install
     (buildpath/"src/github.com/elves").mkpath
     ln_sf buildpath, buildpath/"src/github.com/elves/elvish"
     system "go", "build", "-o", bin/"elvish"
-    system "make", "-C", "src/github.com/elves/elvish", "stub"
```
Comment (dunn) [inline Formula/elvish.rb]: Just for context, what was this `stub` target for?

**Comment by xiaq** [comment]
When executing pipelines Elvish used to put itself in the background, like most other shells do. However, Elvish did not (and still does not) have the concept of subshells, and when executing pipelines that mixed internal commands and external commands, internal commands were executed in the shell's process, now in the background. This meant that they could not get input or signals from the terminal. They could still write to the terminal, provided that the `TTOU` signal was ignored.

This was where the little "stub" came in. It was a little program that runs alongside Elvish, and always kept in the foreground. It used a pipe to relay signals back to the Elvish process, so that internal commands could pretend that they did get signals. (Theoretically it should also relay input, but that was not yet implemented.)

However, the fact that the stub was another program made the building and deploying a little more complicated; it never was an ideal solution. Then I decided to forego the dance of putting the shell in the background before executing a pipeline and back in the foreground afterwards, and made Elvish just keep running in the foreground. So far it seems to have have suited my users' and my needs.

Were the stub deemed necessary again as more interesting use cases arise, it will not be implemented as a separate program, but within the Elvish binary itself. The trick is simple: let Elvish accept a `-stub` flag and if it is present, run as the stub program instead of the normal shell.

**Comment by xiaq** [comment]
Sorry for the wall of text, here is a tl;dr version of it:

It used to be a small program that runs along side Elvish, but the author decided that it complicated the building process, and we could live without it, so he removed it.

**Comment by dunn** [comment]
Awesome, thanks for the detail!

**Comment by dunn** [comment]
Merged as 7029a4c2af72b51b75154e1a15e1b955fe4b3618!

**Pull request closed without merging**